ddtrace/span.py

```diff
@@ -127,6 +128,11 @@ def __init__(
         if not (parent_id is None or isinstance(parent_id, six.integer_types)):
             raise TypeError("parent_id must be an integer")
 
+        self._tracer = None
+        if tracer is not None:
+            # accessing span.tracer logs a deprecation warning
+            self.tracer = tracer  # type: ignore
```

Comment:
Do we need to deprecate the function argument?

What does this error end up looking like?

Reply:
ddtrace/span.py:134: error: Cannot assign to a method  [assignment]

Reply:
I'll update this to `# type: ignore [assignment]`